src/main/resources/META-INF/resources/primefaces/forms/forms.selectonemenu.js

```diff
@@ -1298,10 +1303,105 @@ PrimeFaces.widget.SelectOneMenu = PrimeFaces.widget.DeferredWidget.extend({
             }, 10);
         }
         else {
+            this.renderPanelContentFromHiddenSelect(true);
+
             handleMethod.call(this, event);
         }
     },
 
+    /**
+     * Renders panel content based on hidden select.
+     * @param {boolean} initContentsAndBindItemEvents Call initContents and bindItemEvents after rendering?
+     * @private
+     */
+    renderPanelContentFromHiddenSelect: function(initContentsAndBindItemEvents) {
+         if (this.cfg.renderPanelContentOnClient && this.itemsWrapper.children().length === 0) {
+             var panelContent = '<ul id="' + this.jqId + '_items" class="ui-selectonemenu-items ui-selectonemenu-list ui-widget-content ui-widget ui-corner-all ui-helper-reset" role="listbox">';
+             panelContent += this.renderSelectItems(this.input);
+             panelContent += '</ul>';
+
+             this.itemsWrapper.append(panelContent);
+
+             if (initContentsAndBindItemEvents) {
+                 this.initContents();
+                 this.bindItemEvents();
+             }
+         }
+    },
+
+    /**
+     * Renders Panel-HTML-code for SelectItems.
+     * @private
+     * @param {JQuery} parentItem An parentItem (select, optgroup) for which to render HTML-code.
+     * @return {string} Rendered HTML-code.
+     */
+    renderSelectItems: function(parentItem) {
+        var content = "";
+
+        var opts = parentItem.children("option, optgroup");
+
+        opts.each((i, opt) => {
```

Comment:
This is ES6 syntax is it not?  I think you will have to write it as traditional ES5 javascript.

Reply:
Done. Maybe we should rethink after IE11 is gone whether we relay want to stick to ES5 forever.